fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -318,6 +318,67 @@ private void updateDimensionRow(
     }
 
     /**
+     * Replaces Lucene Index with a new index.
+     *
+     * @param newLuceneIndexPath  The path of the new index
+     */
+    public void replaceIndex(String newLuceneIndexPath) {
+        Directory oldLuceneDirectory = luceneDirectory;
+
+        lock.writeLock().lock();
+        try {
+            // close old dir
+            try {
+                oldLuceneDirectory.close();
+            } catch (IOException e) {
+                String message = ErrorMessageFormat.CANNOT_CLOSE_INDEX_DIR.format(oldLuceneDirectory);
+                LOG.error(message, e);
+            }
+
+            // move the old dir to a different location
+            try {
+                oldLuceneDirectory.rename(luceneIndexPath, luceneIndexPath + ".temp");
```

Comment:
The behavior of rename is unclear, we need to test it before doing anything else with it.
1. Does this keep the current lucene instance available and correct?
2. Does rename changes the directory of which this is pointing to?

@michael-mclawhorn  do you know more about this? looking at the source code... this method is weird I think.

Reply:
Yeah, I'm testing it. Looks like there is a bug from my end while calling `rename`. Investigating...